File: Leetcode/1614. Maximum Nesting Depth of the Parentheses/solution.py
class Solution:
    def maxDepth(self, s: str) -> int:
        max_depth = 0
        open_parentheses = 0

        for char in s:
            if char == '(':
                open_parentheses += 1
                max_depth = max(max_depth, open_parentheses)
            elif char == ')':
                open_parentheses -= 1

        return max_depth

File: Leetcode/1614. Maximum Nesting Depth of the Parentheses/solution.rs
use std::cmp;

impl Solution {
    pub fn max_depth(s: String) -> i32 {
        let mut parentheses_counter = 0;
        let mut max_depth = 0;
        
        for c in s.chars() {
            if c == '(' {
                parentheses_counter += 1;
                max_depth = cmp::max(max_depth, parentheses_counter);
            } else if c == ')'{
                parentheses_counter -= 1;
            }
        }
        
        max_depth
    }
}
